os2phonebook_service/os2phonebook/datastore.py

```diff
@@ -0,0 +1,441 @@
+            }
+        }
+
+        response_data = self.db.search(index=index, body=query)
+        total_size = response_data["hits"]["total"]["value"]
+
+        return int(total_size)
+
+    def get_all_org_units(self) -> List[dict]:
+        """Retrieve all org unit documents from the store
+
+        Returns:
+            List[dict]: An array of org unit documents as dictionaries
+
+        """
+
+        index = "org_units"
+
+        # Calculate total document size
+        total_size = self.get_size(index)
+
+        # Scoped query
+        # Documents returned only contains values for
+        # * uuid
+        # * org unit name
+        # * parent unit uuid
+        query = {
+            "size": total_size,
+            "query": {
+                "match_all": {}
+            },
+            "_source": {
+                "includes": [
+                    "uuid",
+                    "name",
+                    "parent",
+                ]
+            }
+        }
+
+        return_data = self.db.search(index=index, body=query)
+
+        hits = return_data["hits"]["hits"]
+
+        return [
+            unit["_source"]
+            for unit in hits if "_source" in unit
+        ]
+
+    def search(self, search_type, search_value, fuzzy_search=False) -> List[dict]:
+        """High level search method
+
+        Generates search query based on the search type by
+        calling the underlying query method.
+
+        Available search types:
+            * employee_by_name
+            * employee_by_phone
+            * employee_by_email
+            * employee_by_engagement
+            * org_unit_by_name
+
+        Args:
+            search_type (str): Search type (see available types above)
+            search_value (str): Arbitrary search string
+            fuzzy_search (bool): If True, use match_phrase_prefix
+
+        Returns:
+            List[dict]: A list of documents that matched the search
+
+        Example:
+
+            results = client.search(
+                search_type = "employee_by_name",
+                search_value = "Commander Riker",
+                fuzzy_search = True
+            )
+
+        """
+
+        if search_type == "employee_by_name":
+            index, query = self.query_for_employee_by_name(search_value, fuzzy_search)
+
+        elif search_type == "employee_by_phone":
+            index, query = self.query_for_employee_by_phone(search_value, fuzzy_search)
+
+        elif search_type == "employee_by_email":
+            index, query = self.query_for_employee_by_email(search_value, fuzzy_search)
+
+        elif search_type == "employee_by_engagement":
+            index, query = self.query_for_employee_by_engagement(search_value, fuzzy_search)
+
+        elif search_type == "org_unit_by_name":
+            index, query = self.query_for_org_unit_by_name(search_value, fuzzy_search)
+
+        else:
+            raise InvalidSearchType(f"Search type: {search_type} is not available")
```

Comment:
Consider using a simple `search_type -> query_fn` map for this instead of a big if/else structure

Reply:
34446: Conditional structure replaced with a map and a get method function